ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,228 @@
+     * @param contractId The class name of the [Contract] associated with the transaction state.
+     * @param notary The notary that will be used to notarise the specified [ContractState].
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, contractId: String, notary: Party): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @param contractId The class name of the [Contract] associated with the transaction state.
+     * @param notary The notary that will be used to notarise the specified [ContractState].
+     * @param encumbrance The index of an associated, encumbered state, or null if no encumbrance applies to the associated transaction state.
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, contractId: String, notary: Party, encumbrance: Int?): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid from the specified [Instant], tending towards positive infinity.
+     *
+     * @param from The [Instant] from which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowFrom(from: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid until the specified [Instant], tending towards negative infinity.
+     *
+     * @param until The [Instant] until which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowUntil(until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid between the specified [Instant] values.
+     *
+     * @param from The [Instant] from which the transaction time window is valid.
+     * @param until The [Instant] until which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowBetween(from: Instant, until: Instant): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid at the specified [Instant] with the specified tolerance.
+     *
+     * @param midpoint The [Instant] representing the midpoint of the time window.
+     * @param tolerance The [Duration] representing the tolerance of the time window, which is applied before and after the midpoint.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowBetween(midpoint: Instant, tolerance: Duration): UtxoTransactionBuilder
```

Comment:
Not sure we need to support this function?

Reply:
The vararg overloads are useful when you only have to provide a single key. Calling a method with setOf(key) is annoying.